clippy.toml

```diff
@@ -0,0 +1 @@
+allow-mixed-uninlined-format-args = false
```

Comment:
If format args can all be specified inline, then that is great.

I think whether to inline mixed format args is conditioned on what works well for that situation and I think I'd prefer to allow mixed.  This is a case where I'd love a [less than warning](https://internals.rust-lang.org/t/forbid-deny-warn-allow-and-notice/19986) level because I'd like to have these called out with the option to fix them but I don't want it in the same category that gets turned into errors.

Reply:
@epage I think this is a case where the warning is good but the autofix isn't always good. I'm sometimes not a fan of mixing inline and positional, though I think it's fine when there's only *one* positional. So, in practice, sometimes I prefer to fix the mixed case by giving a name to the ones that can't be inlined.

If you'd prefer, I can revert the `clippy.toml` and just submit the lint addition.

Reply:
I think I read this field the wrong and in reading the docs, I'm even more confused.

Re-reading `allow-mixed-uninlined-format-args = true`, it reads to me as "allow some to be inlined and some not to be inlined" and clippy lints if one arg could be inlined even if another couldn't.

Reading the docs

> If allow-mixed-uninlined-format-args is set to false in clippy.toml, the following code will also trigger the lint:
>
> `format!("{} {}", var, 1+2);`
>
> Use instead:
>
> `format!("{var} {}", 1+2);`

That makes it sound like `= false` is the behavior I expect out of `= true`.

Overall, what I want is
- Lint if all can be inlined but aren't
- If one can't be inlined, don't warn on any of the other format args, letting me pick and choose whether I inline some or keep all uninlined within that `format_args!` call

Reply:
`allow-mixed-uninlined-format-args` means "allow" in the sense of `allow(lintname)`; false means to *warn* on mixed-uninlined-format-args, where the "mixed" case is "some can be inlined and some can't".

Reply:
> Overall, what I want is
> 
>     * Lint if all can be inlined but aren't
> 
>     * If one can't be inlined, don't warn on any of the other format args, letting me pick and choose whether I inline some or keep all uninlined within that `format_args!` call

I've updated the PR to have that exact behavior.